riscv/plic.cc

```diff
@@ -156,6 +159,21 @@ bool plic_t::priority_write(reg_t offset, uint32_t val)
   return true;
 }
 
+bool plic_t::pending_read(reg_t offset, uint32_t *val)
+{
+  uint32_t id_word = (offset >> 2);
+
+  if (id_word < num_ids_word) {
+    *val = 0;
+    for (size_t i = 0; i < contexts.size(); i++) {
```

Comment:
Please use a modern C++ loop: `for (auto context: contexts)`

Reply:
Done